opsdroid/connector/matrix/connector.py

```diff
@@ -241,6 +274,17 @@ async def listen(self):  # pragma: no cover
 
             _LOGGER.debug(_("Matrix sync request returned."))
 
+            await self.connection.send_to_device_messages()
+
+            if self.connection.should_upload_keys:
+                await self.connection.keys_upload()
+            if self.connection.should_query_keys:
+                await self.connection.keys_query()
+            if self.connection.should_claim_keys:
```

Comment:
Should this not also be done as part of the initial sync? If so This could be moved into a helper method?